CDP4Composition/ViewModels/ParameterRowControlViewModel.cs

```diff
@@ -0,0 +1,211 @@
+        }
+
+        /// <summary>
+        /// Backing field for <see cref="Switch"/>
+        /// </summary>
+        private string switchField;
+
+        /// <summary>
+        /// Gets or sets the parameter Switch
+        /// </summary>
+        public string Switch
+        {
+            get => this.switchField;
+            set => this.RaiseAndSetIfChanged(ref this.switchField, value);
+        }
+
+        /// <summary>
+        /// Backing field for <see cref="Description"/>
+        /// </summary>
+        private string description;
+
+        /// <summary>
+        /// Gets or sets the parameter Description
+        /// </summary>
+        public string Description
+        {
+            get => this.description;
+            set => this.RaiseAndSetIfChanged(ref this.description, value);
+        }
+
+        /// <summary>
+        /// Backing field for <see cref="ModelCode"/>
+        /// </summary>
+        private string modelCode;
+
+        /// <summary>
+        /// Gets or sets the parameter ModelCode
+        /// </summary>
+        public string ModelCode
+        {
+            get => this.modelCode;
+            set => this.RaiseAndSetIfChanged(ref this.modelCode, value);
+        }
+
+        /// <summary>
+        /// Backing field for <see cref="RowType"/>
+        /// </summary>
+        private string rowType;
+
+        /// <summary>
+        /// Gets or sets the parameter Row Type
+        /// </summary>
+        public string RowType
+        {
+            get => this.rowType;
+            set => this.RaiseAndSetIfChanged(ref this.rowType, value);
+        }
+
+        /// <summary>
+        /// Gets the <see cref="Parameter"/> this view model represents
+        /// </summary>
+        public ParameterOrOverrideBase Parameter { get; private set; }
+
+        /// <summary>
+        /// Gets the Actual Option
+        /// </summary>
+        public Option ActualOption { get; private set; }
+
+        /// <summary>
+        /// Initializes a new instance of <see cref="ParameterRowControlViewModel"/>
+        /// </summary>
+        /// <param name="parameter">The <see cref="Parameter"/> this view model represents</param>
+        /// <param name="actualOption">The actual Option</param>
+        public ParameterRowControlViewModel(ParameterOrOverrideBase parameter, Option actualOption)
+        {
+            this.Parameter = parameter;
+            this.ActualOption = actualOption;
+            this.UpdateProperties();
+        }
+
+        /// <summary>
+        /// Update and set all this view model properties
+        /// </summary>
+        private void UpdateProperties()
+        {
+            var valueSet = this.GetValueSet();
+            this.Name = this.Parameter.ParameterType.Name;
+            this.ShortName = this.Parameter.ParameterType.ShortName;
+            this.Value = $"{valueSet.Published.FirstOrDefault()} [{this.Parameter.Scale.ShortName}]";
```

Comment:
This will throw if `this.GetValueSet()` returns a null. Its unlikely to ever happen, but if for wahtever reason `this.Parameter` is null, but you can ensure.